doc/source/reference/array_api.rst

```diff
@@ -0,0 +1,795 @@
+
+   * - Feature
+     - Type
+     - Notes
+   * - Various functions have been renamed.
+     - **Compatible**
+     - See :ref:`array_api-name-changes`.
+   * - Elementwise functions are only defined for given input type
+       combinations.
+     - **Strictness**
+     - See :ref:`array_api-type-strictness`.
+   * - ``bitwise_left_shift`` and ``bitwise_right_shift`` are only defined for
+       ``x2`` nonnegative.
+     - **Strictness**
+     -
+   * - ``ceil``, ``floor``, and ``trunc`` return an integer with integer
+       input.
+     - **Breaking**
+     - ``np.ceil``, ``np.floor``, and ``np.trunc`` return a floating-point
+       dtype on integer dtype input.
+
+.. _array_api-linear-algebra-differences:
+
+Linear Algebra Differences
+--------------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Feature
+     - Type
+     - Notes
+   * - ``cholesky`` includes an ``upper`` keyword argument.
+     - **Compatible**
+     -
+   * - ``cross`` does not broadcast its arguments.
+     - ???
+     -
+   * - ``cross`` does not allow size 2 vectors (only size 3).
+     - ???
+     -
+   * - ``diagonal`` operates on the last two axes.
+     - **Breaking**
+     -
+   * - ``eigh``, ``qr``, ``slogdet`` and ``svd`` return a named tuple.
+     - **Compatible**
+     - The corresponding ``numpy`` functions return a ``tuple``, with the
+       resulting arrays in the same order.
+   * - New functions ``matrix_norm`` and ``vector_norm``.
+     - **Compatible**
+     - The ``norm`` function has been omitted from the array API and split
+       into ``matrix_norm`` for matrix norms and ``vector_norm`` for vector
+       norms. Note that ``vector_norm`` supports any number of axes, whereas
+       ``np.norm`` only supports a single axis for vector norms.
+   * - ``matrix_rank`` has an ``rtol`` keyword argument instead of ``tol``.
+     - **Compatible**
+     - In the array API, ``rtol`` filters singular values smaller than
+       ``rtol * largest_singular_value``. In ``np.linalg.matrix_rank``,
+       ``tol`` filters singular values smaller than ``tol``. Furthermore, the
+       default value for ``rtol`` is ``max(M, N) * eps``, whereas the default
+       value of ``tol`` in ``np.linalg.matrix_rank`` is ``S.max() *
+       max(M, N) * eps``, where ``S`` is the singular values of the input.
+   * - ``matrix_rank`` does not support 1-dimensional arrays.
+     - **Breaking**
+     -
+   * - New function ``matrix_transpose``.
+     - **Compatible**
+     - Unlike ``np.transpose``, ``matrix_transpose`` only transposes the last
+       two axes. See `the spec definition
+       <https://data-apis.org/array-api/latest/API_specification/generated/signatures.linear_algebra_functions.matrix_transpose.html#signatures.linear_algebra_functions.matrix_transpose>`__
+   * - ``outer`` only supports 1-dimensional arrays.
+     - **Breaking**
+     -
+   * - ``pinv`` has an ``rtol`` keyword argument instead of ``rcond``
+     - **Compatible**
```

Comment:
There was some discussion about this (also for `matrix_rank`). See https://github.com/data-apis/array-api/issues/211 and https://github.com/data-apis/array-api/issues/216.

Reply:
This is breaking indeed, but as breaks go it's as small as they get.

Reply:
Fine to mark as breaking here I'd say, pointing out that the default changes from `1e-15` to `None`, which means `max(M, N) * eps`.

Reply:
I marked these as compatible because the name of the keyword argument also changed.

Reply:
That doesn't matter here right? A `np.linalg.pinv(x)` call would start behaving differently, so that is backwards incompatible. If you were thinking that we'd keep `rcond` _and_ `rtol`, that wouldn't quite make sense.

Reply:
It is an incompatible change and it could at the very least break the reproducibility of workflows (results change and I am not sure how to transition smoothly).  So it should be marked/discussed as such.  In principle it has the potential to be a large change, but I doubt it is in practice.

I would expect that the number of affected users will be rather small and that it should almost never cause _wrong_ results; so it would rather only break reproducibility.  And more often than not, it hopefully will "fix" results also.
Plus, reproducibility may well already be a game of luck for those that would be affected.

Reply:
Ah, you're right of course. The keyword rename can be done compatibly but the default change can't.

Reply:
The same reasoning also applies `matrix_rank`, and to the `stable` flag of `sort` and `argsort`.